### PR TITLE
add MimirContinuousTestFailingOnWrites and MimirContinuousTestFailing…

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Towards: https://github.com/giantswarm/roadmap/issues/3578

As discussed in the above issue, this PR adds 2 alerts based on the mimir's `continous-test` component's metrics so that we get alerted when something is wrong in the read or write path. 

Those 2 alerts are directly taken from upstream mixins as explained in the comments. 

Before merging this PR, I still need to add UTs as well as create a dedicated dashboard for the continous-test component.

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [x] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [x] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml [hunk 1]
```diff
@@ -169,4 +169,38 @@ spec:
         severity: page
         team: atlas
         topic: observability
+    - alert: MimirContinuousTestFailingOnWrites
+      annotations:
+        description: 'Mimir continous-test detected errors in the write path.'
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml:174]:
```suggestion
        dashboard: bdxh7hszfgmbkc/mimir-continous-test
        description: 'Mimir continous-test detected errors in the write path.'
```

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml [hunk 2]
```diff
@@ -169,4 +169,39 @@ spec:
...
+      # Query is based on the following upstream mixin alerting rule : https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled/alerts.yaml#L1097
+      expr: sum by(cluster_id, installation, namespace, pipeline, provider, test) (rate(mimir_continuous_test_writes_failed_total[5m])) > 0
+      for: 1h
+      labels:
+        area: platform
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "true"
+        #TODO dashboard:
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml:186]:
```suggestion
```

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml [hunk 3]
```diff
@@ -169,4 +169,38 @@ spec:
...
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "true"
+        severity: page
+        team: atlas
+        topic: observability
+    - alert: MimirContinuousTestFailingOnReads
+      annotations:
+        description: 'Mimir continous-test detected errors in the write path.'
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml:191]:
```suggestion
        dashboard: bdxh7hszfgmbkc/mimir-continous-test
        description: 'Mimir continous-test detected errors in the write path.'
```

File: helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml [hunk 4]
```diff
@@ -169,4 +169,39 @@ spec:
...
+      # Query is based on the following upstream mixin alerting rule : https://github.com/grafana/mimir/blob/main/operations/mimir-mixin-compiled/alerts.yaml#L1097
+      expr: sum by(cluster_id, installation, namespace, pipeline, provider, test) (rate(mimir_continuous_test_queries_failed_total[5m])) > 0
+      for: 1h
+      labels:
+        area: platform
+        cancel_if_cluster_status_creating: "true"
+        cancel_if_cluster_status_deleting: "true"
+        cancel_if_cluster_status_updating: "true"
+        cancel_if_outside_working_hours: "true"
+        # TODO dashboard:
```
Comment (QuentinBisson) [inline helm/prometheus-rules/templates/platform/atlas/alerting-rules/mimir.rules.yml:203]:
```suggestion
```

**Comment by QuentinBisson** [comment]
This is not a draft anymore, I repeat, this is not a draft anymore

**Review by hervenicol — COMMENTED** [review]
So, we're testing the rate of fails, which is fine.
But the continuous tests also provide a counter of reads and writes. Should we test those as well?

For instance, if we can't write data, we won't get any fails increase. But we also won't get any count of new "continuous test writes", which should probably raise an alert?

**Review by hervenicol — APPROVED** [review]
LGTM.

Note (to self): I'm not happy about the duplicate tests for CAPI. CAPI should now be the default target for tests, and vintage/aws the exception.

**Pull request merged**